backend/shork/src/test/kotlin/instruction/TestSlt.kt

```diff
@@ -0,0 +1,169 @@
+import software.shonk.interpreter.internal.instruction.Dat
+import software.shonk.interpreter.internal.instruction.Seq
+import software.shonk.interpreter.internal.instruction.Slt
+import software.shonk.interpreter.internal.process.Process
+import software.shonk.interpreter.internal.program.Program
+import software.shonk.interpreter.internal.settings.InternalSettings
+
+internal class TestSlt {
+    private val dat = Dat(0, 0, AddressMode.IMMEDIATE, AddressMode.IMMEDIATE, Modifier.A)
+    private val settings = InternalSettings(8000, 1000, dat, 1000, 100)
+    private var shork = InternalShork(settings)
+    private var program = Program("slt", shork)
+    private var process = Process(program, 0)
+
+    @BeforeEach
+    fun setup() {
+        shork = InternalShork(settings)
+        program = Program("slt", shork)
+        process = Process(program, 0)
+    }
+
+    private fun executeSltAndAssertCounter(slt: Slt, expectedCounterIncrement: Int) {
+        slt.execute(process)
+        assertEquals(expectedCounterIncrement, process.programCounter)
+    }
+
+    private fun setupSlt(modifier: Modifier): Slt {
+        val slt = Slt(1, 2, AddressMode.DIRECT, AddressMode.DIRECT, modifier)
+        shork.memoryCore.storeAbsolute(0, slt)
+        return slt
+    }
+
+    private fun setupMemory(aField: Int, bField: Int, address: Int) {
+        val dat = Dat(aField, bField, AddressMode.IMMEDIATE, AddressMode.IMMEDIATE, Modifier.A)
+        shork.memoryCore.storeAbsolute(address, dat)
+    }
+
+    @Test
+    fun `test modifier A fields smaller`() {
+        val slt = setupSlt(Modifier.A)
+        setupMemory(42, 42, 1)
+        setupMemory(69, 69, 2)
+        executeSltAndAssertCounter(slt, 1)
+    }
+
+    @Test
+    fun `test modifier A fields not smaller`() {
+        val slt = setupSlt(Modifier.A)
+        setupMemory(42, 69, 1)
+        setupMemory(42, 69, 2)
+        executeSltAndAssertCounter(slt, 0)
+    }
+
+    @Test
+    fun `test modifier B fields smaller`() {
+        val slt = setupSlt(Modifier.B)
+        setupMemory(42, 42, 1)
+        setupMemory(42, 69, 2)
+        executeSltAndAssertCounter(slt, 1)
+    }
+
+    @Test
+    fun `test modifier B fields not smaller`() {
+        val slt = setupSlt(Modifier.B)
+        setupMemory(42, 69, 1)
+        setupMemory(42, 69, 2)
+        executeSltAndAssertCounter(slt, 0)
+    }
+
+    @Test
+    fun `test modifier AB fields smaller`() {
+        val slt = setupSlt(Modifier.AB)
+        setupMemory(42, 69, 1)
+        setupMemory(42, 69, 2)
+        executeSltAndAssertCounter(slt, 1)
+    }
+
+    @Test
+    fun `test modifier AB fields not smaller`() {
+        val slt = setupSlt(Modifier.AB)
+        setupMemory(42, 69, 1)
+        setupMemory(69, 42, 2)
+        executeSltAndAssertCounter(slt, 0)
+    }
+
+    @Test
+    fun `test modifier BA fields smaller`() {
+        val slt = setupSlt(Modifier.BA)
+        setupMemory(42, 69, 1)
+        setupMemory(42, 69, 2)
+    }
```

Comment:
You missed the check to assert the counter. If you add that, you will also notice that the b-field is in fact bigger than the a-field, so the test condition is not accurate.